### PR TITLE
net-mail/queue-repair: add python3 support

### PR DESCRIPTION
**Pull request by powerman**
Closes: https://bugs.gentoo.org/723250
Package-Manager: Portage-2.3.99, Repoman-2.3.22
Signed-off-by: Aleksandr Efros <powerman-asdf@yandex.ru>

### CODE REVIEW

**Comment by okias** [comment]
Thank you @powerman

File: net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild:4]: Use 7 here.

File: net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Comment (juippis) [inline net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild:6]: 3.8 and 3.9 should work here too, looking at deps?

File: net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+inherit python-single-r1
+
+DESCRIPTION="A toolkit for dealing with the qmail queue directory structure"
+HOMEPAGE="http://pyropus.ca/software/queue-repair/"
+SRC_URI="http://pyropus.ca/software/queue-repair/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86"
+IUSE=""
```
Comment (juippis) [inline net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild:17]: Empty variable can be removed.

File: net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="${PYTHON_DEPS}"
+DEPEND="${RDEPEND}"
```
Comment (juippis) [inline net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild:21]: This is `BDEPEND` here.

File: net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+PATCHES=( "${FILESDIR}/python3.patch" )
```
Comment (juippis) [inline net-mail/queue-repair/queue-repair-0.9.0-r2.ebuild:23]: Please include package name and version in the patch, when this was introduced. So `queue-repair-0.9.0-python3.patch`.

File: net-mail/queue-repair/files/python3.patch [hunk 6]
```diff
@@ -0,0 +1,239 @@
+--- a/queue_repair.py 2003-10-22 23:54:13.000000000 +0800
++++ b/queue_repair.py 2020-01-17 18:21:18.834455813 +0800
+@@ -1,4 +1,4 @@
+-#!/usr/bin/python
++#!/usr/bin/python3
```
Comment (juippis) [inline net-mail/queue-repair/files/python3.patch:5]: You can call `python_fix_shebang` to automatically fix this one. 
https://dev.gentoo.org/~mgorny/python-guide/helper.html

**Comment by powerman** [comment]
I've switched from qmail to postfix and don't use this package anymore.

**Pull request closed without merging**